#  Exception Handling

Java:
In Java, there are 3 excpetion handling components, the try, catch, and finally blocks. To throw an exception, you must use the throw statement and provide it with an exception object. An exception object is a descendent of Throwable. The try block identifies a block of code in which an exception can occur. The try statement should contain at least one catch block or a finally block and may have multiple catch blocks. The catch block identifies a block of code, known as an exception handler, that can handle a particular type of exception. The finally block identifies a block of code that is guaranteed to execute, and is the right place to close files, recover resources, and otherwise clean up after the code enclosed in the try block.

C++:
The language has special constructs, try, throw, and catch, known as exception-handling constructs, to keep error-checking code separate and to reduce redundant checks. A try block surrounds normal code, which is exited immediately if a throw statement executes. A throw statement appears within a try block; if reached, execution jumps immediately to the end of the try block. The code is written so only error situations lead to reaching a throw. A throw statement's syntax is similar to a return statement. A catch clause immediately follows a try block; if the catch was reached due to an exception thrown of the catch clause's parameter type, the clause executes. If no exception is thrown, the code continues normally and all handlers are ignored. Multiple catch statements can be chained together with the first being the most specific, and the last being the most general. If an ellipsis (...) is used as the parameter of catch, that handler will catch any exception no matter what the type of the exception thrown

